traffic_monitor/health/cache_test.go

```diff
@@ -33,9 +33,364 @@ import (
 	"github.com/apache/trafficcontrol/lib/go-tc"
 )
 
+// TestNoMonitoredInterfacesGetVitals assures that GetVitals
+// does not fail even if no interfaces are marked to be monitored
+func TestNoMonitoredInterfacesGetVitals(t *testing.T) {
+	serverID := "no-monitored"
+	fakeRequestTime := time.Now()
+	zeroValueVitals := cache.Vitals{}
+
+	// Interfaces to monitor are marked true (none)
+	tmcm := tc.TrafficMonitorConfigMap{
+		TrafficServer: map[string]tc.TrafficServer{
+			serverID: {
+				Interfaces: []tc.ServerInterfaceInfo{
+					{
+						Name:    "bond0",
+						Monitor: false,
+					},
+					{
+						Name:    "bond1",
+						Monitor: false,
+					},
+					{
+						Name:    "lo",
+						Monitor: false,
+					},
+				},
+			},
+		},
+	}
+
+	// multiple interfaces, plus extra
+	firstResult := cache.Result{
+		ID:            serverID,
+		Error:         nil,
+		Miscellaneous: map[string]interface{}{},
+		Statistics: cache.Statistics{
+			Interfaces: map[string]cache.Interface{
+				"bond0": {
+					Speed:    100000,
+					BytesIn:  570791700709,
+					BytesOut: 4212211168526,
+				},
+				"bond1": {
+					Speed:    100000,
+					BytesIn:  1989352297218,
+					BytesOut: 10630690813,
+				},
+				"lo": {
+					Speed:    0,
+					BytesIn:  181882394,
+					BytesOut: 181882394,
+				},
+				"em5": {
+					Speed:    0,
+					BytesIn:  0,
+					BytesOut: 0,
+				},
+			},
+		},
+		Time:            fakeRequestTime,
+		RequestTime:     time.Second,
+		Vitals:          cache.Vitals{},
+		InterfaceVitals: nil,
+		PollID:          42,
+		PollFinished:    make(chan uint64, 1),
+		PrecomputedData: cache.PrecomputedData{},
+		Available:       true,
+		UsingIPv4:       false,
+	}
+	GetVitals(&firstResult, nil, &tmcm)
+
+	// No interfaces were selected to be monitored so none
+	// should have been added later
+	if len(firstResult.InterfaceVitals) > 0 {
+		t.Errorf("InterfaceVitals map should be empty. expected: %v actual: %v:", 0, len(firstResult.InterfaceVitals))
+	}
+
+	// No interfaces were selected to be monitored so no vitals
+	// should have been calculated
+	if firstResult.Vitals != zeroValueVitals {
+		t.Errorf("Vitals should have zero values. expected: %v actual: %v:", zeroValueVitals, firstResult.Vitals)
+	}
+
+	// multiple interfaces, plus extras
+	secondResult := cache.Result{
```

Comment:
Instead of rewriting the whole struct here, can we just take `firstResult`, copy it into `secondResult` and modify just the required fields?

Reply:
Will do.

Reply:
Done